src/canopy/knowledge_base/reranker/cohere.py

```diff
@@ -0,0 +1,69 @@
+from typing import List, Optional
+
+from canopy.knowledge_base.models import KBQueryResult
+from canopy.knowledge_base.reranker import Reranker
+
+try:
+    import cohere
+except (OSError, ImportError, ModuleNotFoundError):
+    _cohere_installed = False
+else:
+    _cohere_installed = True
+
+
+class CohereReranker(Reranker):
```

Comment:
Just a note: we need to remember to put this as part of the cohere.yaml once we have it